File: codes/easy/2520 Count the Digits That Divide a Number/discussion.js
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function(num) {
    let count = 0;
    let temp = num;

    while(temp > 0) {
        if(num % (temp %10) == 0) count++;
        temp = Math.floor(temp / 10);
    }

    return count;
};

File: codes/easy/2520 Count the Digits That Divide a Number/discussion.py
class Solution:
    def countDigits(self, num: int) -> int:
        count = 0
        temp = num

        while(temp > 0):
            if(num % (temp % 10) == 0):
                count+=1
            
            temp = (temp - (temp % 10)) / 10

        return count